driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,313 @@
+         */
+        public Builder withInclusionThreshold(double errorsPerSecThreshold) {
+            this.errorsThreshold = errorsPerSecThreshold;
+            return this;
+        }
+
+        /**
+         * Provide a class that will allow filtering errors returned by hosts.
+         * <p/>
+         * The {@link ErrorsFilter#shouldAccountForError} method will be called on every error notification the
+         * error tracker associated to the Error aware policy receives. It will then call the error filter, which
+         * will determine whether this error needs to be considered or not.
+         * <p/>
+         * The default implementation will exclude from the error counting, the following exception types :
+         * <ul>
+         *     <li>{@link ReadTimeoutException}</li>
+         *     <li>{@link WriteTimeoutException}</li>
+         *     <li>{@link UnavailableException}</li>
+         *     <li>{@link ReadFailureException}</li>
+         *     <li>{@link WriteFailureException}</li>
+         * </ul>
+         * @param errorsFilter the filter class that the policy will use.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withErrorsFilter(ErrorsFilter errorsFilter) {
+            this.errorsFilter = errorsFilter;
+            return this;
+        }
+
+        /**
+         * Defines the time during which a host is excluded by the tracker.
+         * <p/>
+         * When a host's error rate overpasses the threshold specified in {@link #withInclusionThreshold}, the host
+         * will be considered excluded, and this for a certain period of time defined by this method. After this period,
+         * a previously excluded host will be included again unless/until it reaches the threshold again.
+         * <p/>
+         * If the error tracker, during an update, detects that a currently excluded host continues to overpass the
+         * error rate threshold, this host's exclusion time will be reset to the start.
+         * <p/>
+         * Default value for the retry period is 5 seconds.
+         *
+         * @param retryPeriod the period to use for refreshing the ErrorTracker.
+         * @return this {@link Builder} instance, for method chaining.
+         */
+        public Builder withRetryPeriod(int retryPeriod) {
+            this.retryPeriod = retryPeriod;
+            return this;
+        }
+
+        /**
+         * Creates the {@link ErrorAwarePolicy} instance.
+         * @return the newly created {@link ErrorAwarePolicy}.
+         */
+        public ErrorAwarePolicy build() {
+            return new ErrorAwarePolicy(this);
+        }
+    }
+
+    class PerHostErrorTracker implements LatencyTracker {
+
+        private double errorsThreshold;
+
+        private ConcurrentMap<Host, Meter> hostsMeters = new ConcurrentHashMap<Host, Meter>();
+
+        private ConcurrentMap<Host, Long> exclusions = new ConcurrentHashMap<Host, Long>();
```

Comment:
Well, what's stored is not really the time at which hosts will be resurrected, but the time at which they have been excluded. So `resurrectTimes` doesn't sound right to me